File: Fer/Models/UserProfile.swift
//
//  UserProfile.swift
//  Fer
//
//  Plain `id` (not @DocumentID) — shared with the Watch target, which can't
//  import FirebaseFirestore (no watchOS support).
//

import Foundation

struct UserProfile: Codable, Identifiable {
    var id: String?
    var displayName: String = ""
    var email: String = ""
    var weightUnit: WeightUnit = .lb
    var createdAt: Date = Date()

    enum WeightUnit: String, Codable, CaseIterable {
        case lb, kg
        var label: String { rawValue }
    }
}

File: android/app/src/main/java/com/harshbshah/fer/data/model/UserProfile.kt
package com.harshbshah.fer.data.model

import com.google.firebase.firestore.Exclude
import java.util.Date

data class UserProfile @JvmOverloads constructor(
    @get:Exclude @set:Exclude
    var id: String? = null,

    var displayName: String = "",
    var email: String = "",
    var weightUnit: WeightUnit = WeightUnit.lb,
    var createdAt: Date = Date()
) {
    enum class WeightUnit {
        lb, kg;

        val label: String get() = name
    }
}
